application/src/test/java/com/sanction/thunder/resources/UserResourceTest.java

```diff
@@ -35,15 +38,16 @@ public void testPostNullUser() {
 
   @Test
   public void testPostUserInvalidEmail() {
-    PilotUser user = new PilotUser("badEmail", "password", "", "", "");
+    PilotUser user = new PilotUser(badEmail, "password", "", "", "");
     Response response = resource.postUser(key, user);
 
     assertEquals(Response.Status.BAD_REQUEST, response.getStatusInfo());
   }
 
   @Test
   public void testPostUserDatabaseDown() {
-    when(usersDao.insert(user)).thenThrow(new DatabaseException(DatabaseError.DATABASE_DOWN));
+    when(usersDao.insert(any(PilotUser.class)))
+        .thenThrow(new DatabaseException(DatabaseError.DATABASE_DOWN));
```

Comment:
Just curious, why'd you have to change this to `any(PilotUser.class)`?

Reply:
Well now that I have a fully mocked user I can change this back. At first I had to try and match the same user that was passed in and since it had a random number I couldn't pass in a hard coded user.

Reply:
Never mind you need it. When you pass in a user to the post method it gets filled with a random token when inserted into the DB. When I try to return an object for "usersDao.insert(user)" I have no way of knowing the random token so it fails every time since my mocked user object doesn't match what is actually returned.

Reply:
Makes sense, thanks!